pkg/catalog/catalog.go

```diff
@@ -115,13 +117,25 @@ const (
 
 var ErrUnknownDiffType = errors.New("unknown graveler difference type")
 
+type ctxCloser struct {
+	fn context.CancelFunc
```

Comment:
I may prefer:
```suggestion
	close context.CancelFunc
```
describes what it does, not its type.